modules/sso.php

```diff
@@ -461,13 +462,109 @@ private function should_hide_login_form() {
 		return apply_filters( 'jetpack_remove_login_form', get_option( 'jetpack_sso_remove_login_form', false ) );
 	}
 
+	/**
+	 * Outputs the Jetpack SSO button and description as well as the toggle link
+	 * for switching between Jetpack SSO and default login.
+	 */
 	function login_form() {
-		$classes = '';
+		$site_name = get_bloginfo( 'name' );
+		if ( ! $site_name ) {
+			$site_name = get_bloginfo( 'url' );
+		}
 
-		if ( $this->should_hide_login_form() ) {
-			$classes .= ' forced-sso';
+		$display_name = ! empty( $_COOKIE[ 'jetpack_sso_wpcom_name' . COOKIEHASH ] )
+			? $_COOKIE[ 'jetpack_sso_wpcom_name' . COOKIEHASH ]
+			: false;
+		$gravatar = ! empty( $_COOKIE[ 'jetpack_sso_wpcom_gravatar' . COOKIEHASH ] )
+			? $_COOKIE[ 'jetpack_sso_wpcom_gravatar' . COOKIEHASH ]
+			: false;
+
+		?>
+		<div id="jetpack-sso-wrap">
+			<?php if ( $display_name && $gravatar ) : ?>
+				<div id="jetpack-sso-wrap__user">
+					<img width="72" height="72" src="<?php echo esc_html( $gravatar ); ?>" />
+
+					<h2>
+						<?php
+							_e(
+								sprintf( 'Log in as <span>%s</span>', esc_html( $display_name ) ),
+								'jetpack'
+							);
+						?>
+					</h2>
+				</div>
+
+			<?php endif; ?>
+
+
+			<div id="jetpack-sso-wrap__action">
+				<?php echo $this->build_sso_button( array(), 'is_primary' ); ?>
+
+				<?php if ( $display_name && $gravatar ) : ?>
+					<a class="jetpack-sso-wrap__reauth" href="<?php echo $this->build_sso_button_url( array( 'reauth' => '1' ) ); ?>">
+						<?php esc_html_e( 'Log in as a different WordPress.com user', 'jetpack' ); ?>
+					</a>
+				<?php else : ?>
+					<p>
+						<?php
+							_e(
```

Comment:
@ebinnion same than above regarding the `sprintf`, but in this case the entire string can be escaped so:
```php
echo esc_html(
    sprintf( __( 'You can now save time spent logging in by connecting your WordPress.com account to %s.', 'jetpack' ), esc_html( $site_name ) ) )
);
```